src/chocolatey/infrastructure.app/services/NugetService.cs

```diff
@@ -71,36 +71,31 @@ public void list_noop(ChocolateyConfiguration config)
                                 ));
         }
 
-        public ConcurrentDictionary<string, PackageResult> list_run(ChocolateyConfiguration config, bool logResults = true)
+        public IEnumerable<PackageResult> list_run(ChocolateyConfiguration config, bool logResults)
         {
-            var packageResults = new ConcurrentDictionary<string, PackageResult>();
-
-            var packages = NugetList.GetPackages(config, _nugetLogger).ToList();
-
-            foreach (var package in packages.or_empty_list_if_null())
+            foreach (var package in NugetList.GetPackages(config, _nugetLogger))
             {
+                var pkg = package; // for lamda access

```

Comment:
Yeah, I made that change in the next pull request which touches these files :-( I tried and tried to get move that change into this commit but I couldn't get git/diff to cooperate.